### PR TITLE
Exposing aws_v4_authentication configuration in global external alb

### PR DESCRIPTION
**Pull request by okguru1**
Exposing `aws_v4_authentication` configuration in global external alb

---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/net-lb-app-ext/backend-service.tf [hunk 1]
```diff
@@ -257,6 +257,12 @@ resource "google_compute_backend_service" "default" {
     content {
       client_tls_policy = ss.value.client_tls_policy
       subject_alt_names = ss.value.subject_alt_names
+      aws_v4_authentication {
```
Comment (juliocc) [inline modules/net-lb-app-ext/backend-service.tf:260]: you probably want a dynamic block here

File: modules/net-lb-app-ext/variables-backend-service.tf [hunk 2]
```diff
@@ -123,8 +123,13 @@ variable "backend_service_configs" {
     security_settings = optional(object({
       client_tls_policy = string
       subject_alt_names = list(string)
-    }))
-  }))
+      aws_v4_authentication = object({
```
Comment (juliocc) [inline modules/net-lb-app-ext/variables-backend-service.tf:126]: the new field should be optional

**Review by juliocc — CHANGES_REQUESTED** [review]
Left a few comments below.

**Comment by okguru1** [comment]
> Left a few comments below.

Fixed

**Pull request merged**